### PR TITLE
Oscar fhd

### PR DESCRIPTION
**Pull request by aelanman**
Includes new files for running FHD on the Oscar cluster at Brown, and one small change to vis_simulate allowing it to handle complex uv arrays.

[And a change to handle single-polarization cases]

### CODE REVIEW

File: Observations/firstpass_versions_wrapper.pro [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+case version of
+  'ael_test_0': begin
+	max_model_sources=1000
+	dft_threshold=1
+	no_ps=0
+  end
+endcase
+   
+if version EQ 'nb_pytest' then begin
```
Comment (isullivan) [inline Observations/firstpass_versions_wrapper.pro]: If `version` is ever 'nb_pytest' then the above `case` statement will crash. You might want to add an `else` to the case statement, or add other versions that you expect to use

File: instrument_config/paper_beam_setup_gain.pro [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FUNCTION paper_beam_setup_gain,obs,antenna,file_path_fhd=file_path_fhd,$
+FUNCTION paper_beam_setup_gain,obs,antenna,file_path_fhd=file_path_fhd,jdate_use=jdate_use,$
```
Comment (isullivan) [inline instrument_config/paper_beam_setup_gain.pro]: jdate_use isn't actually used in this file. To catch keywords that may be passed in but are not used, add _extra=extra as the final keyword.

File: simulation/vis_simulate.pro [hunk 3]
```diff
@@ -40,7 +40,10 @@ FUNCTION vis_simulate,obs,status_str,psf,params,jones,skymodel,file_path_fhd=fil
     beam_arr=beam_image_cube(obs,psf, n_freq_bin = n_freq,/square)
     for freq_i=0,n_freq-1 do begin
       beam2_xx_image[*,*, freq_i] = Temporary(*beam_arr[0,freq_i])
-      beam2_yy_image[*,*, freq_i] = Temporary(*beam_arr[1,freq_i])
+    ;  beam2_yy_image[*,*, freq_i] = Temporary(*beam_arr[1,freq_i])
+      if n_pol gt 1 THEN beam2_yy_image[*,*, freq_i] = Temporary(*beam_arr[1,freq_i]) $
+      else beam2_yy_image[*,*,freq_i] = beam2_xx_image[*,*,freq_i]
+
```
Comment (isullivan) [inline simulation/vis_simulate.pro]: To keep style consistent, please use multi-line if ... then ... else statements, in the form
IF n_pol GT 1 THEN BEGIN
    do stuff
ENDIF ELSE BEGIN
   do other stuff
ENDELSE

Comment (isullivan) [inline simulation/vis_simulate.pro]: (GitHub strips the spaces I added for indentation, so please do use indentation inside if statements!)

File: simulation/vis_simulate.pro [hunk 4]
```diff
@@ -153,7 +156,15 @@ FUNCTION vis_simulate,obs,status_str,psf,params,jones,skymodel,file_path_fhd=fil
...
-        FOR pol_i=0,n_pol-1 DO *model_uvf_arr[pol_i]+=Rebin(*source_model_uv_arr[pol_i],dimension,elements,n_freq,/sample)
+	IF size(*source_model_uv_arr[0],/type) eq 6 then begin     ;; If array is complex
+	        FOR pol_i=0,n_pol-1 DO BEGIN
+			j = Complex(0,1)
+			*model_uvf_arr[pol_i]+=Rebin(Real_part(*source_model_uv_arr[pol_i]),dimension,elements,n_freq,/sample)$
+							+j*(Rebin(imaginary(*source_model_uv_arr[pol_i]),dimension,elements,n_freq,/sample))
+		ENDFOR
+	ENDIF ELSE BEGIN
+	        FOR pol_i=0,n_pol-1 DO *model_uvf_arr[pol_i]+=Rebin(*source_model_uv_arr[pol_i],dimension,elements,n_freq,/sample)
+	ENDELSE
```
Comment (isullivan) [inline simulation/vis_simulate.pro]: Use the rebin_complex function instead, from fhd_utils

**Comment by bhazelton** [comment]
Looks like the issues Ian had are fixed, so I'm merging.

**Pull request merged**